yarn-project/end-to-end/src/e2e_multiple_accounts_1_enc_key.test.ts

```diff
@@ -0,0 +1,161 @@
+
+import {
+  createNewAccount,
+  expectUnencryptedLogsFromLastBlockToBe,
+  expectsNumOfEncryptedLogsInTheLastBlockToBe,
+  setup,
+} from './utils.js';
+
+describe('e2e_multiple_accounts_1_enc_key', () => {
+  let aztecNode: AztecNodeService | undefined;
+  let aztecRpcServer: AztecRPC;
+  const wallets: Wallet[] = [];
+  const accounts: AztecAddress[] = [];
+  let logger: DebugLogger;
+
+  let zkTokenAddress: AztecAddress;
+
+  const initialBalance = 987n;
+  const numAccounts = 3;
+
+  beforeEach(async () => {
+    ({ aztecNode, aztecRpcServer, logger } = await setup(0));
+
+    const encryptionPrivateKey = randomBytes(32);
+    for (let i = 0; i < numAccounts; i++) {
+      logger(`Deploying account contract ${i}/3...`);
+      const signingPrivateKey = randomBytes(32);
+      const createWallet = async (address: AztecAddress, useProperKey: boolean) =>
+        new StoredKeyAccountContract(address, useProperKey ? signingPrivateKey : randomBytes(32), await Schnorr.new());
+
+      const schnorr = await Schnorr.new();
+      const signingPublicKey = schnorr.computePublicKey(signingPrivateKey);
+      const constructorArgs = [
+        Fr.fromBuffer(signingPublicKey.subarray(0, 32)),
+        Fr.fromBuffer(signingPublicKey.subarray(32, 64)),
+      ];
+
+      const { wallet, address } = await createNewAccount(
+        aztecRpcServer,
+        SchnorrMultiKeyAccountContractAbi,
+        constructorArgs,
+        encryptionPrivateKey,
+        true,
+        createWallet,
+      );
+      wallets.push(wallet);
+      accounts.push(address);
+    }
+    logger('Account contracts deployed');
+
+    // Verify that all accounts use the same encryption key
+    const encryptionPublicKey = await generatePublicKey(encryptionPrivateKey);
+    for (let i = 0; i < numAccounts; i++) {
+      const accountEncryptionPublicKey = await aztecRpcServer.getPublicKey(accounts[i]);
+      expect(accountEncryptionPublicKey).toEqual(encryptionPublicKey);
+    }
+
+    logger(`Deploying ZK Token...`);
+    const tx = ZkTokenContract.deploy(aztecRpcServer, initialBalance, accounts[0]).send();
+    const receipt = await tx.getReceipt();
+    zkTokenAddress = receipt.contractAddress!;
+    await tx.isMined(0, 0.1);
+    const minedReceipt = await tx.getReceipt();
+    expect(minedReceipt.status).toEqual(TxStatus.MINED);
+    logger('ZK Token deployed');
+  }, 100_000);
+
+  afterEach(async () => {
+    await aztecNode?.stop();
+    if (aztecRpcServer instanceof AztecRPCServer) {
+      await aztecRpcServer?.stop();
+    }
+  });
+
+  const expectBalance = async (userIndex: number, expectedBalance: bigint) => {
+    const wallet = wallets[userIndex];
+    const owner = accounts[userIndex];
+
+    // First wait until the corresponding RPC server has synchronised the account
+    const isUserSynchronised = async () => {
+      return await wallet.isAccountSynchronised(owner);
+    };
+    await retryUntil(isUserSynchronised, owner.toString(), 5);
```

Comment:
Curious if we actually need this, since we only have 1 rpc server on this test

Reply:
I looked into the code and it really doesn't seem to be needed here because when tx is mined it is ensured that all the notes were already processed. I removed the check in [0b4aa5a](https://github.com/AztecProtocol/aztec-packages/pull/1137/commits/0b4aa5a2cbdd3c4d05d5cedc7d74829c60d41b35). Thanks